debounce/simple_debouncer_test.go

```diff
@@ -0,0 +1,56 @@
+package debounce
+
+import (
+	"runtime"
+	"testing"
+)
+
+// TestDebouncer tests that the debouncer allows only one function to execute at a time
+func TestSimpleDebouncer(t *testing.T) {
+	d := NewSimpleDebouncer()
+	executions := 0
+	startedCh := make(chan struct{}, 1)
+	doneCh := make(chan struct{}, 1)
+	// Function to increment executions
+	fn := func() {
+		<-startedCh // Simulate work
+		executions++
+		<-doneCh // Simulate work
+	}
+	t.Run("Case 1", func(t *testing.T) {
+		// Case 1: Normal single execution
+		startedCh <- struct{}{}
+		doneCh <- struct{}{}
+		d.Debounce(fn)
+		// We expect that the function has only executed once due to debouncing
+		if executions != 1 {
+			t.Errorf("Expected function to be executed only once, but got %d executions", executions)
+		}
+	})
+	executions = 0
+	t.Run("Case 2", func(t *testing.T) {
+		// Case 2: Debounce the function multiple times at row when body is started
+		go d.Debounce(fn)
+		startedCh <- struct{}{}
+		// Switch to `go d.Debounce(fn)` goroutine
+		runtime.Gosched()
```

Comment:
Probably this one needs to ne replaced with `waitTillChannelIsEmpty(startedCh)`

Reply:
this is weird, both versions worked locally, but in the CI they are failing